### ISSUE TITLE
Leap day/af issue

### ISSUE THREAD

**Issue by merfish** [Ouranosinc/xclim#685]
The following example has ERA-5 and CMIP6 input data, where we removed leap days and assign a `dayofyear` coordinate. Following the training and adjusting steps, the output data preserves the `dayofyear` coordinate and the variable retains the 1-365 (no leap days) metric. However, the `af` calculated following the training step has a 1-366 `dayofyear` coordinate with data in each day, which is odd given that the input datasets don't have data for the additional day. 

The ERA-5 data and notebook are attached in the zip file.
[xclim_issue.zip](https://github.com/Ouranosinc/xclim/files/6273862/xclim_issue.zip)

**Comment by aulemahal** [comment]
Thanks for the complete working example! Makes debugging a lot easier.

**Explanation**:

The "dayofyear" coordinate in the training dataset is not computed by xclim, rather by xarray, somewhere in our `Grouper.apply`. Using `time.dayofyear` as the grouper argument will result in the same as `da.groupby('time.dayfoyear').apply(func)`, just that our `Grouper` object is a bit more complex and allows fancier things. Nonetheless, the "day of year" of each time element is computed individually, i.e.: it is independent of the real number of "days" in the current year of data. `xarray` represents dates using either `numpy.datetime64` or `cftime.DatetimeYYYY` objects (where YYYY stands for the calendar type).

All this to explain that on a leap year, December 31st _is_ the 366th day, whether February 29th is present or not in the data. I am thus not surprised that the training dataset had 366 elements along `dayofyear`.

**Solution:**

Instead of simply removing February 29th, you also need to convert the time coordinate from the current "standard" calendar (xclim calls it "default" as it is stored in numpy objects) to a "noleap" calendar, where you indeed always have 365 days per year. Lucky day: there is an helper in xclim.

I suggest replacing:
```python3
ref = remove_leap_days(ref_da)
hist = remove_leap_days(hist_da)
sim = remove_leap_days(sim_da)
```
by
```python3
from xclim.core.calendar import convert_calendar

ref = convert_calendar(ref_da, 'noleap')
hist = convert_calendar(hist_da, 'noleap')
sim = convert_calendar(sim_da, 'noleap')
```
This drops the incompatible days (Feb 29) and converts each elements to a `cftime.DatetimeNoLeap` object.

More info : https://xclim.readthedocs.io/en/stable/api.html#xclim.core.calendar.convert_calendar

**Comment by aulemahal** [comment]
However, another solution should have worked. Instead of using `grouper = sdba.Grouper("time.dayofyear", window=31)`, I expected `grouper = sdba.Grouper("dayofyear", window=31)` to work in your case, since you added a `dayofyear` auxiliary coordinate with the wanted values. It didn't work and that is a bug of xclim!

**Comment by merfish** [comment]
Converting the calendar worked, thanks for the suggestion!
